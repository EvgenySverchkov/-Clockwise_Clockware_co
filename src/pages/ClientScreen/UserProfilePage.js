import React from "react";
import { Switch, Route } from "react-router-dom";

import NavBtnGroup from "../../components/UserProfile/NavBtnGroup";
import UserOrdersList from "../../components/UserProfile/UserOrdersList";
import UserSettings from "../../components/UserProfile/UserSettings";

const UserProfilePage = () => {
  if (localStorage.getItem("token")) {
    return (
      <>
        <NavBtnGroup />
        <Switch>
          <Route
            path="/client/userProfile/userOrders"
            component={UserOrdersList}
          />
          <Route
            path="/client/userProfile/userSettings"
            component={UserSettings}
          />
        </Switch>
      </>
    );
  } else {
    return (
      <div className="text-center display-4">
        Please, log into your account!
      </div>
    );
  }
};

export default UserProfilePage;
