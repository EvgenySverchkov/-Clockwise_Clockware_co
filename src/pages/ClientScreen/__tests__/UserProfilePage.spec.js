import React from "react";
import {create, act} from "react-test-renderer";
import {Provider} from "react-redux";
import configMockStore from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";

import UserProfilePage from "../UserProfilePage";
import Context from "../../../ContextComponent";

const mockStore = configMockStore();

describe("Test of <UserProfilePage />", ()=>{
    const store = mockStore({
        ordersReducer:{
            currentOrder: {}
        },
        ordersReducer: {
            userOrders: [{
                id: 1,
                name: "Name",
                email: "test@test.com",
                size: "middle",
                town: "Town",
                date: "2021-01-01",
                time: "13:00",
                masterId: 5,
                endTime: "15:00"
            }]
        },
        userProfileReducer: {
            editUserDataIsLoad: false
        }
    });
    const mockContextValue = {
        openErrorWindowWithMsg: jest.fn(),
        openWarningTooltip: jest.fn(),
        warningTooltipMsg: "Warning"
    }
    const Component = ({path}) => (
        <MemoryRouter initialEntries={[path]}>
            <Context.Provider value = {mockContextValue}>
                <Provider store = {store}>
                    <UserProfilePage />
                </Provider>
            </Context.Provider>
        </MemoryRouter>
    );
    beforeEach(()=>global.fetch = jest.fn(()=>Promise.resolve({json: jest.fn(()=>Promise.resolve())})))
    it("When path is '/client/userProfile/userOrders' and user logged (field 'token' in localStorage is true)", async ()=>{
        global.localStorage.__proto__.getItem = ()=>true;
        let component
        await act(async ()=>{
            component = await create(
                <Component path = {"/client/userProfile/userSettings"}/>
            );
        })
        expect(component.toJSON()).toMatchSnapshot();
    });
    
    it("When path is '/client/userProfile/userSettings' and user logged (field 'token' in localStorage is true)", ()=>{
        global.localStorage.__proto__.getItem = ()=>JSON.stringify({
            email: "test@mail.com",
            name: "Name"
        });
        
        const component = create(<Component path={'/client/userProfile/userSettings'}/>);
        expect(component.toJSON()).toMatchSnapshot();
    })
    
    it("When path doesn't logged (field 'token' in localStorage is false)", ()=>{
        global.localStorage.__proto__.getItem = ()=>false;
        const component = create(<Component path={"/client/userProfile"}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});