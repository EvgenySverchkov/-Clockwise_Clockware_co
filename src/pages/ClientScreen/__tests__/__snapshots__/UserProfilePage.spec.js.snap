// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <UserProfilePage /> When path doesn't logged (field 'token' in localStorage is false) 1`] = `
<div
  className="text-center display-4"
>
  Please, log into your account!
</div>
`;

exports[`Test of <UserProfilePage /> When path is '/client/userProfile/userOrders' and user logged (field 'token' in localStorage is true) 1`] = `
Array [
  <div
    className="btn-group col-12 p-0 mb-2"
    role="group"
  >
    <a
      className="btn btn-secondary"
      href="/client/userProfile/userOrders"
      onClick={[Function]}
      type="button"
    >
      Orders
    </a>
    <a
      className="btn btn-secondary"
      href="/client/userProfile/userSettings"
      onClick={[Function]}
      type="button"
    >
      Settings
    </a>
    <a
      className="btn btn-danger"
      href="/"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>,
  <form
    className="mt-4 row justify-content-center"
    onBlur={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-group row text-center text-sm-left col-12 col-md-10 col-lg-8"
    >
      <label
        className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label font-weight-bold"
        htmlFor="userMail"
      >
        Your email
      </label>
      <div
        className="col-sm-8 col-md-8"
      >
        <input
          aria-describedby={null}
          className="form-control"
          id="userMail"
          name="email"
          onBlur={[Function]}
          onChange={null}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          readOnly={true}
          required={true}
          title="Warning"
          type="email"
        />
      </div>
    </div>
    <div
      className="form-group row text-center text-sm-left col-12 col-md-10 col-lg-8"
    >
      <label
        className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label font-weight-bold"
        htmlFor="userName"
      >
        Your name
      </label>
      <div
        className="col-sm-8 col-md-8"
      >
        <input
          aria-describedby={null}
          className="form-control"
          id="userName"
          name={null}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          required={true}
          title="Warning"
          type="text"
        />
      </div>
    </div>
    <div
      className="row justify-content-sm-center col-12"
    >
      <input
        className="btn btn-primary col-12 col-sm-4 mt-3"
        type="submit"
        value="Edit profile"
      />
    </div>
  </form>,
]
`;

exports[`Test of <UserProfilePage /> When path is '/client/userProfile/userSettings' and user logged (field 'token' in localStorage is true) 1`] = `
Array [
  <div
    className="btn-group col-12 p-0 mb-2"
    role="group"
  >
    <a
      className="btn btn-secondary"
      href="/client/userProfile/userOrders"
      onClick={[Function]}
      type="button"
    >
      Orders
    </a>
    <a
      className="btn btn-secondary"
      href="/client/userProfile/userSettings"
      onClick={[Function]}
      type="button"
    >
      Settings
    </a>
    <a
      className="btn btn-danger"
      href="/"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>,
  <form
    className="mt-4 row justify-content-center"
    onBlur={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-group row text-center text-sm-left col-12 col-md-10 col-lg-8"
    >
      <label
        className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label font-weight-bold"
        htmlFor="userMail"
      >
        Your email
      </label>
      <div
        className="col-sm-8 col-md-8"
      >
        <input
          aria-describedby={null}
          className="form-control"
          id="userMail"
          name="email"
          onBlur={[Function]}
          onChange={null}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          readOnly={true}
          required={true}
          title="Warning"
          type="email"
          value="test@mail.com"
        />
      </div>
    </div>
    <div
      className="form-group row text-center text-sm-left col-12 col-md-10 col-lg-8"
    >
      <label
        className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label font-weight-bold"
        htmlFor="userName"
      >
        Your name
      </label>
      <div
        className="col-sm-8 col-md-8"
      >
        <input
          aria-describedby={null}
          className="form-control"
          id="userName"
          name={null}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          required={true}
          title="Warning"
          type="text"
          value="Name"
        />
      </div>
    </div>
    <div
      className="row justify-content-sm-center col-12"
    >
      <input
        className="btn btn-primary col-12 col-sm-4 mt-3"
        type="submit"
        value="Edit profile"
      />
    </div>
  </form>,
]
`;
